spec/webauthn/public_key_credential_spec.rb

```diff
@@ -92,5 +92,68 @@
         }.to raise_error(WebAuthn::ChallengeVerificationError)
       end
     end
+
+    context "when clientExtensionResults is received" do
+      let(:public_key_credential) do
+        WebAuthn::PublicKeyCredentialWithAttestation.new(
+          type: type,
+          id: id,
+          raw_id: raw_id,
+          client_extension_outputs: { "appid" => "true" },
+          response: attestation_response
+        )
+      end
+
+      it "works" do
+        expect(public_key_credential.verify(challenge)).to be_truthy
+
+        expect(public_key_credential.client_extension_outputs).to eq({ "appid" => "true" })
+      end
+    end
+
+    context "when authentication extension input" do
+      context "is not received" do
+        let(:attestation_response) do
+          allow_any_instance_of(WebAuthn::FakeAuthenticator::AuthenticatorData)
+            .to receive(:extensions)
+            .and_return(nil)
+
+          response = client.create(challenge: raw_challenge)["response"]
+
+          WebAuthn::AuthenticatorAttestationResponse.new(
+            attestation_object: response["attestationObject"],
+            client_data_json: response["clientDataJSON"]
+          )
+        end
+
+        it "works" do
+          expect(public_key_credential.verify(challenge)).to be_truthy
+
+          expect(public_key_credential.authenticator_extension_outputs).to be_nil
+        end
+      end
+
+      context "is received" do
+        let(:attestation_response) do
+          allow_any_instance_of(WebAuthn::FakeAuthenticator::AuthenticatorData)
+            .to receive(:extensions)
+            .and_return({ "fakeExtension" => "fakeExtensionValue" })
```

Comment:
I know that this is not needed given that those are the default extensions but I like to be explicit declare the values that will be latter tested against

Reply:
@bdewater somehow I managed to delete all our discussion's thread while trying to delete the message from above in order to move it to another thread 🙁 
Please feel free to re-create the conversation if you have the comments in your email or something (I disabled those github's email notifications) or post the messages in the Pivotal Tracker and I will do it for sure.
Sorry for that... 😕

Reply:
No worries, I sort of remember the comment :)

- You can pass `nil` as a value to the `client_extension_outputs` argument, it will work without falling back to the default value.
- I believe `nil` is correct based on my reading of [CTAP 2.0](https://fidoalliance.org/specs/fido-v2.0-ps-20190130/fido-client-to-authenticator-protocol-v2.0-ps-20190130.html#authenticatorMakeCredential). I would however not expect to see the equivalent of an empty Ruby hash to be returned in the real world.